Detectors/TRD/base/src/TrackletTransformer.cxx

```diff
@@ -0,0 +1,127 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "DetectorsBase/GeometryManager.h"
+#include "TRDBase/TrackletTransformer.h"
+#include "TMath.h"
+
+using namespace o2::trd;
+using namespace o2::trd::constants;
+
+TrackletTransformer::TrackletTransformer()
+{
+  o2::base::GeometryManager::loadGeometry();
+  mGeo = Geometry::instance();
+  mGeo->createPadPlaneArray();
+  mGeo->createClusterMatrixArray();
+}
+
+void TrackletTransformer::loadPadPlane(int hcid)
+{
+  int detector = hcid / 2;
+  int stack = mGeo->getStack(detector);
+  int layer = mGeo->getLayer(detector);
+  mPadPlane = mGeo->getPadPlane(layer, stack);
+}
+
+float TrackletTransformer::calculateX()
+{
+  // report x at 5mm below the cathode plan to minimize error propogration from tracklet fit and driftV calibration
+  float driftHeight = mGeo->cdrHght();
+  float x = driftHeight - 0.5;
```

Comment:
The 5mm below the cathode plane should be a class member, and we should probably provide setters and getters for it. I think we have discussed a few options, and we should be flexible enough to use all of them:

 - at timebin 0 (this comes from the online tracklets)
 - at the anode wires (explained by Ole)
 - at the cathode wires (the drift velocity did not have time to mess around with the hits, and the tracklet should extrapolate back to here - but I have not finally thought that through)
 - somewhere in the drift region, to minimize the correlation between y and dy. 

I think the last three are straightforward: the member variable gives the position within the chamber. We could reserve e.g. -100 for the case of TB=0. But maybe that case is not even needed.

BTW: where is x=0? Is this at the cathode wires? Or at the radiator?

Reply:
Wasnt 100% certain what you had in mind, but let me know what you think of my implementation. I was a bit unsure about what lengths to use here. There are a few member variables in `GeometryBase.h`. There is one called `ANODEPOS`, but this does not seem to be measured from the start of the drift region so I used a different number instead.

As far as I understand, x=0 is at the start of the drift region.